src/AggregateRepository.EventStore/EventStoreAggregateRepository.cs

```diff
@@ -75,14 +70,7 @@ public T GetAggregateFromRepository<T>(object aggregateId, int version = int.Max
             var streamName = StreamNameForAggregateId(aggregateId);
             var events = ReadFromStream(streamName, version);
 
-            try
-            {
-                return CreateAndRehydrateAggregateAsync<T>(events, version).Result;
-            }
-            catch (AggregateException ex) when (ex.InnerException is AggregateVersionException)
-            {
-                throw ex.InnerException;
-            }
```

Comment:
Why did we change this logic? Do we not want to bubble the AggregateVersionExceptions up from the InnerException?

Reply:
Probably the very similar names confused you `AggregateVersionException` (ours) vs `AggregateException` from the framework.

We were previously using `.Result` on l80, which meant when `CreateAndRehydrate` threw an `AggregateVersionException` it would end up wrapped in an `AggregateException` and we had to extract the real exception from `InnerException`.

That doesn't happen any more now we are using async/await so the `AggregateVersionException` bubbles up directly.